popup.js

```diff
@@ -0,0 +1,423 @@
+                   <a href="#"
+                  ><img src="${portfolioData[0].image}" class="desktop-only" alt="portfolio-1"
+                /></a>
+                
+                </div>
+                <div class="left-col">
+                <div class="text-side">
+                <p class="desktop-only" >
+                  ${portfolioData[0].details}
+                </p>
+                <p class="mobile-only" >
+                  ${portfolioData[7].details2}
+                </p>
+                </div>
+                <div class="links-side">
+                <ul>
+                  <li class="work-tools">
+                    <button type="button" class="tools desktop-only"> ${portfolioData[0].workTools[0]}</button>
+                    <button type="button" class="tools desktop-only">${portfolioData[0].workTools[1]}</button>
+                    <button type="button" class="tools desktop-only">${portfolioData[0].workTools[2]}</button>
+                    <button type="button" class="tools desktop-only">${portfolioData[0].workTools[3]}</button>
+                    <button type="button" class="tools desktop-only">${portfolioData[0].workTools[4]}</button>
+                    <button type="button" class="tools desktop-only">${portfolioData[0].workTools[5]}</button>
+                    <button type="button" class="tools mobile-only">${portfolioData[0].workTools[0]}</button>
+                    <button type="button" class="tools mobile-only">${portfolioData[0].workTools[1]}</button>
+                    <button type="button" class="tools mobile-only">${portfolioData[0].workTools[2]}</button>
+                  </li>
+                </ul>
+               <div class='live-btn'>
+                  <div class= "btn-1">
+                  <button type="button" class="pop-btn">
+                    <a href="${portfolioData[7].liveLink}" target="_blank">See Live</a>
+                    <span><img src="${portfolioData[7].liveIcon}" alt='live'>
+                  </button>
+                </div>
+                  
+                  <div class= "btn-2">
+                  <button type="button" class="pop-btn">
+                  <a href="${portfolioData[7].sourceLink}" target="_blank">See Source</a>
+                  <span><img src="${portfolioData[7].sourceIcon}" alt='gitHub'>
+                  </button>
+                  </div>
+                </div>
+              </div>
+              </div>
+              </div>`;
+
+const overlay = document.querySelector('.back-pop');
+function popDisplay() {
+  dynamic.style.display = 'block';
+  overlay.style.display = 'flex';
+}
+
+const display = document.querySelectorAll('.primary-btn');
+Array.from(display).forEach((element) => {
+  element.addEventListener('click', popDisplay);
+});
+
+const closeModal = document.querySelector('.x-icon');
+
+const popup = document.querySelector('.popup');
+
+function closePopup() {
+  popup.style.display = 'none';
+  overlay.style.display = 'none';
+}
```

Comment:
- [ ] I can see you have static data in your popup, kindly ensure that the data on the popup matches the data of the project from the project array. For instance, if a user clicks on the first project, the popup will display information of the first project, and the same for the second project and all other projects.

Reply:
At the moment, the pop-up displays the same data, regardless of which card the user wants more information on.

We need to dynamically display the data from the array of objects for each and every card separately. 
Here the loops are coming into use.

Please [find here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Loops_and_iteration) how we should use the loops in JavaScript.

This refactoring is mandatory in order to get this milestone approved.